partiql-catalog/src/lib.rs

```diff
@@ -0,0 +1,230 @@
+use crate::call_defs::CallDef;
+use partiql_value::Value;
+use std::borrow::Cow;
+
+use std::collections::HashMap;
+use std::error::Error;
+use std::fmt::Debug;
+use std::sync::atomic::{AtomicU64, Ordering};
+use thiserror::Error;
+use unicase::UniCase;
+
+pub mod call_defs;
+
+pub trait Extension: Debug {
+    fn name(&self) -> String;
+    fn load(&self, catalog: &mut dyn Catalog) -> Result<(), Box<dyn Error>>;
+}
+
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Copy, Clone, Hash)]
+struct CatalogId(u64);
+
+impl From<u64> for CatalogId {
+    fn from(value: u64) -> Self {
+        CatalogId(value)
+    }
+}
+
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Copy, Clone, Hash)]
+struct EntryId(u64);
+
+impl From<u64> for EntryId {
+    fn from(value: u64) -> Self {
+        EntryId(value)
+    }
+}
```

Comment:
So what's the idea behind having an Id associated with the Catalog and Entry?

Reply:
Eventually, `Catalog`s will likely contain other `Catalog`s. This is one way to model nesting of namespaces with lookup. In such a case, an `EntryId` will not be globally unique, but when combined with the id of the `Catalog` it came from, it should be.